test/OpenFeature.Contrib.ConfigCat.Test/ConfigCatProviderTest.cs

```diff
@@ -0,0 +1,101 @@
+        [InlineAutoData(1, "empty", "empty", ErrorType.TypeMismatch)]
+        [InlineAutoData(false, "empty", "empty", ErrorType.TypeMismatch)]
+        public Task GetStringValue_ForFeature_ReturnExpectedResult(object value, string defaultValue, string expectedValue, ErrorType expectedErrorType, string sdkKey)
+        {
+            return ExecuteResolveTest(value, defaultValue, expectedValue, expectedErrorType, sdkKey, (provider, key, def) => provider.ResolveStringValue(key, def));
+        }
+
+        [Theory]
+        [InlineAutoData(1, 2, 1, ErrorType.None)]
+        [InlineAutoData(1.0, 0, 0, ErrorType.TypeMismatch)]
+        [InlineAutoData("false", 0, 0, ErrorType.TypeMismatch)]
+        [InlineAutoData(false, 0, 0, ErrorType.TypeMismatch)]
+        public Task GetIntValue_ForFeature_ReturnExpectedResult(object value, int defaultValue, int expectedValue, ErrorType expectedErrorType, string sdkKey)
+        {
+            return ExecuteResolveTest(value, defaultValue, expectedValue, expectedErrorType, sdkKey, (provider, key, def) => provider.ResolveIntegerValue(key, def));
+        }
+
+        [Theory]
+        [AutoData]
+        public async Task GetStructureValue_ForFeature_ReturnExpectedResult(string sdkKey)
+        {
+            const string jsonValue = "{ \"key\": \"value\" }";
+            var defaultValue = new Value(jsonValue);
+            var configCatProvider = new ConfigCatProvider(sdkKey,
+                options => { options.FlagOverrides = BuildFlagOverrides(("example-feature", defaultValue)); });
+
+            var result = await configCatProvider.ResolveStructureValue("example-feature", defaultValue);
+
+            Assert.Equal(defaultValue.AsString, result.Value.AsString);
+            Assert.Equal("example-feature", result.FlagKey);
+            Assert.Equal(ErrorType.None, result.ErrorType);
+        }
+
+        private static async Task ExecuteResolveTest<T>(object value, T defaultValue, T expectedValue, ErrorType expectedErrorType, string sdkKey, Func<ConfigCatProvider, string, T, Task<ResolutionDetails<T>>> resolveFunc)
+        {
+            var configCatProvider = new ConfigCatProvider(sdkKey,
+                options => { options.FlagOverrides = BuildFlagOverrides(("example-feature", value)); });
+
+            var result = await resolveFunc(configCatProvider, "example-feature", defaultValue);
+
+            Assert.Equal(expectedValue, result.Value);
+            Assert.Equal("example-feature", result.FlagKey);
+            Assert.Equal(expectedErrorType, result.ErrorType);
+        }
+
+        private static FlagOverrides BuildFlagOverrides(params (string key, object value)[] values)
+        {
+            var dictionary = new Dictionary<string, object>();
+            foreach (var (key, value) in values)
+            {
+                dictionary.Add(key, value);
+            }
+
+            return FlagOverrides.LocalDictionary(dictionary, OverrideBehaviour.LocalOnly);

```

Comment:
This definitely makes testing this provider easy!

Reply:
Kudos to the ConfigCat team.